libeos-payg/service.c

```diff
@@ -184,15 +253,35 @@ manager_new_cb (GObject      *source_object,
 {
   g_autoptr(GTask) task = G_TASK (user_data);
   EpgService *self = EPG_SERVICE (g_task_get_source_object (task));
+  g_autoptr(EpgProvider) provider = NULL;
   g_autoptr(GError) local_error = NULL;
 
-  self->provider = epg_manager_new_finish (result, &local_error);
-  if (self->provider == NULL)
+  provider = epg_manager_new_finish (result, &local_error);
+  if (provider == NULL)
     {
       g_task_return_error (task, g_steal_pointer (&local_error));
       return;
     }
 
+  epg_service_startup_complete (self, task, provider);
+}
+
+/**
+ * epg_service_startup_complete:
+ * @self:
+ * @task: (transfer none):
+ * @provider: (transfer full):
+ *
```

Comment:
Needs documentation.

Reply:
Does this internal function need any more than this? I only wrote a docstring at all to have a place to put the transfer annotations. I am happy to just delete it.